src/populate_persons.sql

```diff
@@ -0,0 +1,45 @@
+
+TRUNCATE TABLE person CASCADE;
+TRUNCATE TABLE cast_member CASCADE;
+TRUNCATE TABLE crew_member CASCADE;
+TRUNCATE TABLE job_category CASCADE;
+
+INSERT INTO person (name, birth_date, death_date, "description", imdb_id)
+SELECT 
+    LEFT(primaryname, 50) AS name,  -- Truncate name to 50 characters to prevent errors
+    CASE WHEN birthyear = '' THEN NULL ELSE TO_DATE(birthyear, 'YYYY') END AS birth_date,
+    CASE WHEN deathyear = '' THEN NULL ELSE TO_DATE(deathyear, 'YYYY') END AS death_date,
+    primaryprofession AS description,
+    nconst AS imdb_id
+FROM original.name_basics;
+
+WITH job_roles AS (
+    SELECT DISTINCT category
+    FROM original.title_principals
+)
+INSERT INTO job_category (name)
+SELECT category
+FROM job_roles;
```

Comment:
Is the with statement needed here? can't you just do the following?

```suggestion
INSERT INTO job_category (name)
SELECT DISTINCT category 
FROM original.title_principals
```

Reply:
No it is not.